python/paddle/fluid/tests/unittests/test_tensor_fill_diagonal_tensor_.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle.fluid as fluid
+import paddle.nn.functional as F
+import unittest
+import numpy as np
+import six
+import paddle
+
+
+class TensorFillDiagTensor_Test(unittest.TestCase):
```

Comment:
1. No tests with the case `dim1 > dim2`.
2. Why not use `OpTest`?

Reply:
done, add both